src/WCSimWCAddDarkNoise.cc

```diff
@@ -0,0 +1,254 @@
+	    //Int_t hit_cylLoc;
+	    // TODO: need to change the format of hit_pos to G4ThreeVector
+	    // and change hit_rot to G4RotationMatrix
+	    
+	    WCSimPmtInfo* pmtinfo = (WCSimPmtInfo*)pmts->at( noise_pmt -1 );
+	    hit_pos[0] = 10*pmtinfo->Get_transx();
+	    hit_pos[1] = 10*pmtinfo->Get_transy();
+	    hit_pos[2] = 10*pmtinfo->Get_transz();
+	    hit_rot[0] = pmtinfo->Get_orienx();
+	    hit_rot[1] = pmtinfo->Get_orieny();
+	    hit_rot[2] = pmtinfo->Get_orienz();
+	    //hit_cylLoc = pmtinfo->Get_cylocation();
+	    G4RotationMatrix pmt_rotation(hit_rot[0], hit_rot[1], hit_rot[2]);
+	    G4ThreeVector pmt_position(hit_pos[0], hit_pos[1], hit_pos[2]);
+	    ahit->SetRot(pmt_rotation);
+	    ahit->SetPos(pmt_position);
+	    ahit->SetTime(PMTindex[noise_pmt],current_time);
+	    pe = WCPMT->rn1pe();
+	    ahit->SetPe(PMTindex[noise_pmt],pe);
+	    //Added this line to increase the totalPe by 1
+	    ahit->AddPe(current_time);
+	    WCHCPMT->insert(ahit);
+	    PMTindex[noise_pmt]++;
+	    //	    list[ noise_pmt ] = WCHCPMT->entries();
+	    number_entries ++;
+	    list[ noise_pmt ] = number_entries; // Add this PMT to the end of the list
+#ifdef WCSIMWCADDDARKNOISE_VERBOSE
+	    G4cout << "WCSimWCAddDarkNoise::AddDarkNoiseBeforeDigi Added NEW DIGI with dark noise hit at time " << current_time << " to PMT " << noise_pmt << G4endl;
+#endif
+	  }
+	else{
+	  (*WCHCPMT)[ list[noise_pmt]-1 ]->AddPe(current_time);
+	  pe = WCPMT->rn1pe();
+	  (*WCHCPMT)[ list[noise_pmt]-1 ]->SetPe(PMTindex[noise_pmt],pe);
+	  (*WCHCPMT)[ list[noise_pmt]-1 ]->SetTime(PMTindex[noise_pmt],current_time);
+	  (*WCHCPMT)[ list[noise_pmt]-1 ]->AddParentID(-1);
+	  PMTindex[noise_pmt]++;
+#ifdef WCSIMWCADDDARKNOISE_VERBOSE
+	  G4cout << "WCSimWCAddDarkNoise::AddDarkNoiseBeforeDigi Added to exisiting digi a dark noise hit at time " << current_time << " to PMT " << noise_pmt << G4endl;	  
+#endif
+	}
+		
+      }
+    
+    delete [] PMTindex;
+    return;
+ }
+
+
+
+void WCSimWCAddDarkNoise::FindDarkNoiseRanges(WCSimWCDigitsCollection* WCHCPMT, float width) {
+  //Loop over all Hits and assign a time window around each hit
+  //store these in the ranges vector as pairs
+  for (int g=0; g<WCHCPMT->entries(); g++){
+    for (int gp=0; gp<(*WCHCPMT)[g]->GetTotalPe(); gp++){
+      float time = (*WCHCPMT)[g]->GetTime(gp);
+      //lets assume a 5us window.  So we centre this on the hit time.
+      //t1 is the lower limit of the window.
+      float t1=time - width/2.;
+      float t2=time + width/2.;
+      //if t1 is negative set to 0
+      //if(t1 < 0.)
+      //t1=0.;
+      ranges.push_back(std::pair<float, float>(t1, t2));
+    }
+  }
+  
+  //we need to ensure that the ranges found above are sorted first
+  //for the algorithm below to work
+  //output are pairs stored in the result vector
+  sort(ranges.begin(),ranges.end());
+  std::vector<std::pair<float, float> >::iterator it = ranges.begin();
+  std::pair<float, float> current = *(it)++;
```

Comment:
This while loop is really a for loop, so it should be coded as one. Or did I miss something?